### DISCUSSION TITLE
Adding encompassingEncounter section to C-CDA?

### DISCUSSION THREAD

**Discussion by TronActive** [synthetichealth/synthea#1231] [Ideas]
Has there been any discussion about adding some new sections to C-CDA template? I would like to at least add encompassingEncounter section. I would also like to add Goals, Family and Medical History, and payers' section templateId 2.16.840.1.113883.10.20.22.2.18.

I have no idea what it would take to do this.

**Comment by jawalonoski** [comment]
Apologies for the late response -- your question somehow fell through the cracks.

We don't have any plans to update the C-CDA templates except for bug fixes.

The template format is [Apache Freemarker](https://freemarker.apache.org). If you want to add a new section, you have to modify the templates ([files](https://github.com/synthetichealth/synthea/tree/master/src/main/resources/templates/ccda)). For instance:
https://github.com/synthetichealth/synthea/blob/8f363ad28a5b35df9ea685b68da61d78a60e478e/src/main/resources/templates/ccda/encounters.ftl#L5-L11


If you know C-CDA, adding the new section to the Freemarker template should be pretty straight-forward.

Depending on what the contents of that particular section are, it is possible you might want to create some custom data fields that you can reference inside the Freemarker template in order to make the export easier. If so, you could edit the Java to add those custom data fields. For example,

https://github.com/synthetichealth/synthea/blob/8f363ad28a5b35df9ea685b68da61d78a60e478e/src/main/java/org/mitre/synthea/export/CCDAExporter.java#L123-L132

Hope this helps.
